client/src/Movies/Movie.js

```diff
@@ -1,67 +1,50 @@
-import React, { Component } from 'react';
+import React from 'react';
 import axios from 'axios';
-
-export default class Movie extends Component {
-  constructor(props) {
-    super(props);
+import MovieCard from './MovieCard'
+export default class Movie extends React.Component {
+  constructor(props){
+    super(props)
     this.state = {
-      movie: null
+      movie: null,
     };
   }
 
   componentDidMount() {
-    // change this line to grab the id passed on the URL
-    const id = 1;
-    this.fetchMovie(id);
+    this.fetchMovie(this.props.match.params.id)
+  }
+
+  componentWillReceiveProps(newProps){
```

Comment:
interesting use of lifecycle methods, could you explain your workflow and thought process here?